### DISCUSSION TITLE
Support extra data structures (eg linked list)  as built-in types

### DISCUSSION THREAD

**Discussion by SleepyRoy** [vlang/v#11639] [Ideas]
Now V has very useful arrays and (hash?) maps.
Should V support other commonly used data structures as built-in types? For example linked list, stack, queue, heap (priority queue),  binary search tree, graph, and even data table (like dataframe in Python)?

**Comment by dhonx** [comment]
We already discussed this in Discord. And as far as I know, Alex and some other members not want this. although this is very useful.

**Comment by ka-weihe** [comment]
@dhonx I think you might have misunderstood that discussion. 

`Someone wrote`: "Yea and i don't want to see little packages everywhere just to make a basic functionality reuseable like npm (if you used node.js you get the meme) small things like that should be builtin."

`Alex replied`: "exactly, that's the philosophy of V
  all essential stuff is included
  no more 150 strings, json, web framework libs
  so yeah all these data structures will be built in
  even though you should almost never use linked lists "
